### ISSUE TITLE
unable to build Tf lite .whl file for raspberry pi 0

### ISSUE THREAD

**Issue by Athuliva** [tensorflow/tensorflow#59037]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Build/Install

### Source

source

### Tensorflow Version

tf lite

### Custom Code

No

### OS Platform and Distribution

Ubuntu 20.04.5 LTS

### Python version

3.9

### Current Behaviour?

```shell
I am trying to generate a .whl file tensorflow lite for raspberry pi 0 architecture, using ARM cross compilation, by following the article https://www.tensorflow.org/lite/guide/build_cmake_pip. 
`tensorflow/lite/tools/pip_package/build_pip_package_with_cmake.sh rpi0 ` After running this code, it throughs error in the process `cmake --build . --verbose -j 2 -t _pywrap_tensorflow_interpreter_wrapper`
```

### Standalone code to reproduce the issue

```shell
`tensorflow/lite/tools/pip_package/build_pip_package_with_cmake.sh rpi0 `
```

### Relevant log output

```shell
file included from /home/athul/workspace/abdu/tensorflow_src/tensorflow/lite/tools/cmake/toolchains/arm-rpi-linux-gnueabihf/x64-gcc-6.5.0/arm-rpi-linux-gnueabihf/arm-rpi-linux-gnueabihf/sysroot/usr/include/limits.h:123:0,
                 from /home/athul/workspace/abdu/tensorflow_src/tensorflow/lite/tools/pip_package/gen/tflite_pip/python3/cmake_build/FXdiv-source/include/fxdiv.h:12,
                 from /home/athul/workspace/abdu/tensorflow_src/tensorflow/lite/tools/pip_package/gen/tflite_pip/python3/cmake_build/pthreadpool-source/src/portable-api.c:13:
/usr/include/limits.h:145:17: error: missing binary operator before token "("
 #if __GLIBC_USE (IEC_60559_BFP_EXT_C2X)
                 ^
make[3]: *** [pthreadpool/CMakeFiles/pthreadpool.dir/build.make:76: pthreadpool/CMakeFiles/pthreadpool.dir/src/portable-api.c.o] Error 1
make[3]: Leaving directory '/home/athul/workspace/abdu/tensorflow_src/tensorflow/lite/tools/pip_package/gen/tflite_pip/python3/cmake_build'
make[2]: *** [CMakeFiles/Makefile2:5727: pthreadpool/CMakeFiles/pthreadpool.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
In file included from /home/athul/workspace/abdu/tensorflow_src/tensorflow/lite/tools/cmake/toolchains/arm-rpi-linux-gnueabihf/x64-gcc-6.5.0/arm-rpi-linux-gnueabihf/arm-rpi-linux-gnueabihf/sysroot/usr/include/limits.h:123:0,
                 from /home/athul/workspace/abdu/tensorflow_src/tensorflow/lite/tools/pip_package/gen/tflite_pip/python3/cmake_build/abseil-cpp/absl/base/config.h:52,
                 from /home/athul/workspace/abdu/tensorflow_src/tensorflow/lite/tools/pip_package/gen/tflite_pip/python3/cmake_build/abseil-cpp/absl/base/attributes.h:37,
                 from /home/athul/workspace/abdu/tensorflow_src/tensorflow/lite/tools/pip_package/gen/tflite_pip/python3/cmake_build/abseil-cpp/absl/base/log_severity.h:21,
                 from /home/athul/workspace/abdu/tensorflow_src/tensorflow/lite/tools/pip_package/gen/tflite_pip/python3/cmake_build/abseil-cpp/absl/base/log_severity.cc:15:
/usr/include/limits.h:145:17: error: missing binary operator before token "("
 #if __GLIBC_USE (IEC_60559_BFP_EXT_C2X)
                 ^
make[3]: *** [_deps/abseil-cpp-build/absl/base/CMakeFiles/absl_log_severity.dir/build.make:76: _deps/abseil-cpp-build/absl/base/CMakeFiles/absl_log_severity.dir/log_severity.cc.o] Error 1
make[3]: Leaving directory '/home/athul/workspace/abdu/tensorflow_src/tensorflow/lite/tools/pip_package/gen/tflite_pip/python3/cmake_build'
make[2]: *** [CMakeFiles/Makefile2:2030: _deps/abseil-cpp-build/absl/base/CMakeFiles/absl_log_severity.dir/all] Error 2
make[2]: Leaving directory '/home/athul/workspace/abdu/tensorflow_src/tensorflow/lite/tools/pip_package/gen/tflite_pip/python3/cmake_build'
make[1]: *** [CMakeFiles/Makefile2:1283: CMakeFiles/_pywrap_tensorflow_interpreter_wrapper.dir/rule] Error 2
make[1]: Leaving directory '/home/athul/workspace/abdu/tensorflow_src/tensorflow/lite/tools/pip_package/gen/tflite_pip/python3/cmake_build'
make: *** [Makefile:195: _pywrap_tensorflow_interpreter_wrapper] Error 2
```
</details>

**Comment by mohantym** [comment]
Hi @Athuliva !

Could you check the document specific  to linux runtime like [raspberry pi](https://www.tensorflow.org/lite/guide/python#:~:text=The%20tflite_runtime%20package%20is%20a,want%20to%20do%20is%20execute%20.).

Thank you!

**Comment by Athuliva** [comment]
the os architecture of the raspberry pi 0 that I tried is `armv6l`

**Comment by mohantym** [comment]
Ok @Athuliva !

Thanks for the update.
Could you check this [CMake documentation for Armv6.](https://www.tensorflow.org/lite/guide/build_cmake_arm#build_for_raspberry_pi_zero_armv6) 

Thank you!

**Comment by mcguirepr89** [comment]
Hello -- I am also unable to build the .whl for armv6l when attempting to cross-compile following the instructions for [CMake documentation for Armv6](https://www.tensorflow.org/lite/guide/build_cmake_arm#build_for_raspberry_pi_zero_armv6).

My logs show the exact same failure as @Athuliva.

Also, the aforementioned documentation is missing the `mkdir ~/toolchains` call present in the instructions for the other architectures, so that copy/paste section requires attention.

Please let me know what steps could be helpful or are required to troubleshoot this issue further.

My very best regards,
Patrick

**Comment by Athuliva** [comment]
@mohantym I followed [CMake documentation for Armv6.](https://www.tensorflow.org/lite/guide/build_cmake_arm#build_for_raspberry_pi_zero_armv6).  Running the cmake generated some files i am not sure what to do next. Is it possible to generate .whl file from those generated files?

**Comment by mohantym** [comment]
@Athuliva @mcguirepr89 !

Thanks for the update on armv6 wheels.   
You can check instructions  for [Tensorlfow/Tflite-micro](https://github.com/tensorflow/tflite-micro) raspberry pi installations as an alternate . 

Thank you!

**Comment by mcguirepr89** [comment]
Wow!!! I can just look through the docs!!! What a help!!!

**Comment by arfaian** [comment]
Building the whl for armv6 should be fixed with 63c028fe7ba4d82c0ad9156533acaddaad377049 and was broken for some time due to an outdated toolchain. @Athuliva you can sync with the HEAD and run the following command:

```sh
make -C tensorflow/lite/tools/pip_package docker-build TENSORFLOW_TARGET=rpi0
```

**Comment by gaikwadrahul8** [comment]
Hi, @Athuliva 

Apologize for the delayed response and could please refer above [comment](https://github.com/tensorflow/tensorflow/issues/59037#issuecomment-1386272230) and please use below command for Build TensorFlow Lite Python Wheel Package and check whether your issue is resolving or not ? Also please refer our official documentation [Ref-1](https://www.tensorflow.org/lite/guide/build_cmake_arm#build_for_raspberry_pi_zero_armv6), [Ref-2](https://www.tensorflow.org/lite/guide/build_cmake_pip) which may help you to resolve your issue

`make -C tensorflow/lite/tools/pip_package docker-build TENSORFLOW_TARGET=rpi0`

If issue still persists please let us know or Could you please confirm if this issue is resolved for you ? Please feel free to close the issue if it is resolved ? Thank you!
